lib/internal/source_map/source_map_cache.js

```diff
@@ -205,14 +204,26 @@ function dataFromUrl(sourceURL, sourceMappingURL) {
 // from. This allows translation from byte offset V8 coverage reports,
 // to line/column offset Source Map V3.
 function lineLengths(content) {
-  // We purposefully keep \r as part of the line-length calculation, in
-  // cases where there is a \r\n separator, so that this can be taken into
-  // account in coverage calculations.
-  return ArrayPrototypeMap(RegExpPrototypeSymbolSplit(/\n|\u2028|\u2029/, content), (line) => {
-    return line.length;
-  });
+  const contentLength = content.length;
+  const output = [];
+  let lineLength = 0;
+  for (let i = 0; i < contentLength; i++, lineLength++) {
+    const codePoint = StringPrototypeCodePointAt(content, i);
+
+    // We purposefully keep \r as part of the line-length calculation, in
+    // cases where there is a \r\n separator, so that this can be taken into
+    // account in coverage calculations.
+    // codepoints for \n (new line), \u2028 (line separator) and \u2029 (paragraph separator)
+    if (codePoint === 10 || codePoint === 0x2028 || codePoint === 0x2029) {
+      output.push(lineLength)
```

Comment:
Using the non-primordial version due to `ArrayPrototypePush` being listed as having perf issues — https://github.com/nodejs/node/blob/main/doc/contributing/primordials.md#primordials-with-known-performance-issues